rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3124,6 +3221,56 @@ static rmw_ret_t rmw_send_response_request(
   }
 }
 
+enum class client_present_t
+{
+  ERROR,  // an error occurred when checking
+  MAYBE,  // reader not matched, writer still present
+  YES,    // reader matched
+  GONE    // neither reader nor writer
+};
+
+static bool check_client_service_endpoint(
+  const dds_builtintopic_endpoint_t * ep,
+  const std::string key, const std::string needle)
+{
+  if (ep != nullptr) {
+    std::string clientid;
+    get_user_data_key(ep->qos, key, clientid);
+    return clientid == needle;
+  }
+  return false;
+}
+
+static client_present_t check_for_response_reader(
+  const CddsCS & service,
+  const rmw_request_id_t & reqid)
+{
+  std::vector<dds_instance_handle_t> rds, wrs;
+  if (get_matched_endpoints(service.pub->enth, dds_get_matched_subscriptions, rds) < 0 ||
+    get_matched_endpoints(service.sub->enth, dds_get_matched_publications, wrs) < 0)
+  {
+    RMW_SET_ERROR_MSG("rmw_send_response: failed to get reader/writer matches");
+    return client_present_t::ERROR;
+  }
+  const std::string needle = request_id_writer_guid_to_string(reqid);
+  // if we have matched this client's reader, all is well
+  for (auto rdih : rds) {
```

Comment:
@eboasson nit: 
```suggestion
  for (auto & rdih : rds) {
```

Reply:
given that `dds_instance_handle_t` is just a 64-bit integer, I don't understand why using a reference would be an improvement